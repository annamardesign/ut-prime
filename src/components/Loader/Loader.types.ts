import {WithStyles, withStyles, createStyles, Theme} from '@material-ui/core/styles';
import React from 'react';

export interface Props extends React.HTMLAttributes<HTMLDivElement> {
    className?: string;
    message?: string;
    open?: boolean;
}

const styles = (theme: Theme) => createStyles({
    '@keyframes rotate': {
        from: {
            transform: 'rotate(0deg)',
            msTransform: 'rotate(0deg)',
            MozTransform: 'rotate(0deg)',
            WebkitTransform: 'rotate(0deg)',
            OTransform: 'rotate(0deg)'
        },

        to: {
            transform: 'rotate(360deg)',
            msTransform: 'rotate(360deg)',
            MozTransform: 'rotate(360deg)',
            WebkitTransform: 'rotate(360deg)',
            OTransform: 'rotate(360deg)'
        }
    },
    loaderContainer: {
        position: 'fixed',
        top: '0',
        left: '0',
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        height: '100%',
        width: '100%',
        zIndex: 9998
    },
    overlay: {
        position: 'absolute',
        top: '0',
        left: '0',
        height: '100%',
        width: '100%',
        backgroundColor: theme.palette.background.default,
        opacity: 0.7
    },
    loader: {
        height: 35,
        width: 35,
        borderRadius: '100%',
        border: '3px solid #4096fd',
        borderBottomColor: 'transparent',
        animation: '$rotate 0.8s linear infinite'
    },
    message: {
        marginTop: 20,
        fontSize: 16,
        fontColor: '#373a3c',
        fontFamily: 'Roboto, Arial',
        zIndex: 1
    }
});

export const Styled = withStyles(styles);
export type StyledType = React.FC<Props & WithStyles<typeof styles>>
