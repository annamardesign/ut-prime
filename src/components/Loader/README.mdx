# Loader

Internal component, no do not use it directly

It is used to check an animated loader during data
fetching.
