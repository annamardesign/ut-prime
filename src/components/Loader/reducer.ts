import immutable from 'immutable';

const defaultState = immutable.fromJS({
    open: false,
    requests: 0,
    message: 'Loading, please wait...'
});

export default (state = defaultState, action) => {
    if (action.suppressPreloadWindow) {
        return state;
    }
    if (action.methodRequestState === 'requested') {
        return state
            .set('open', true)
            .set('requests', state.get('requests') + 1)
            .set('message', action.prefetchWindowText || defaultState.get('message'));
    } else if (action.methodRequestState === 'finished') {
        const requests = state.get('requests');
        if (requests <= 1) {
            return state
                .set('open', false)
                .set('requests', 0);
        } else {
            return state
                .set('requests', requests - 1);
        }
    }
    return state;
};
