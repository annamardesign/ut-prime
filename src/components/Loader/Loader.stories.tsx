import React from 'react';
import { withReadme } from 'storybook-readme';

import Wrap from '../test/wrap';

// @ts-ignore: md file and not a module
import README from './README.md';
import Loader from './index';

export default {
    title: 'Loader',
    component: Loader,
    decorators: [withReadme(README)]
};

const state = {
    preloadWindow: {
        toJS: () => ({open: true})
    }
};

export const Basic: React.FC<{}> = () => <Wrap state={state}>
    <Loader>Hello Loader</Loader>
</Wrap>;
