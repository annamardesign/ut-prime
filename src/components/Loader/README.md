# Button

## How to use

```jsx
import Loader from 'ut-front-devextreme/core/Loader';

<Loader onClick={handlerOnClick}>Hello Loader</Loader>;
```

## Props

- **className** - class applied to root `Loader` html element.

| propName  | propType | defaultValue | isRequired |
| --------- | -------- | ------------ | ---------- |
| className | string   |              | no         |
