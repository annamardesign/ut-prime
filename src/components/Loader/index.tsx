import React from 'react';
import clsx from 'clsx';
import { useSelector } from 'react-redux';

import Text from '../Text';

import { useStyles, ComponentProps } from './Loader.types';
import { State } from '../Store/Store.types';

const Loader: ComponentProps = ({ className, message, open }) => {
    const classes = useStyles();
    const loader = useSelector((state: State) => state?.loader);
    return (
        ((loader && loader.open) || open) && <div className={clsx(classes.loaderContainer, className)}>
            <div className={classes.overlay} />
            <div className={clsx(classes.loader, className)} />
            <div className={classes.message}><Text>{message || loader.message}</Text></div>
        </div>
    ) || null;
};

export default Loader;
