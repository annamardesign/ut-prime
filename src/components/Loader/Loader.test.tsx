import React from 'react';

import { render } from '../test';
import Loader from './index';

describe('<Loader />', () => {
    it('render equals snapshot', () => {
        const {
            getByTestId
        } = render(<Loader />, {
            loader: {
                open: true,
                requests: 0,
                message: 'Loading...'
            }
        });
        expect(getByTestId('ut-front-test')).toMatchSnapshot();
    });
});
