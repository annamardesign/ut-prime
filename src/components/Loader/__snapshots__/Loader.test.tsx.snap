// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> render equals snapshot 1`] = `
<div
  data-testid="ut-front-test"
>
  <div
    class="loaderContainer-0-2-2"
  >
    <div
      class="overlay-0-2-3"
    />
    <div
      class="loader-0-2-4"
    />
    <div
      class="message-0-2-5"
    >
      Loading...
    </div>
  </div>
</div>
`;
